qir/qat/TransformationRulesPass/TransformationRulesPass.cpp

```diff
@@ -758,8 +756,11 @@ namespace quantum
         }
 
         // Applying rule set
+
         if (config_.shouldTransformExecutionPathOnly())
         {
+            // TODO(tfr): This implementation path lacks backwards replacement (see else section)
```

Comment:
Could you elaborate what you mean "for each pass"? The current code is inside the TransformationRulePass. You could add this pass after every other pass if you like. Regarding wether the pass throws, in general it should not but it could if we decided it should. However, this part is experimental functionality that needs a revision anyway and which for must cases have a completely different behaviour to the code in the else statement. It would need the backwards replacement to work properly though.

Reply:
I am not fully following the explanation. Given that this is an experimental functionality, let's catch up offline another time and move forward for this PR.